Content/Items/Forest/Armors.SlimePrinceMinion.cs

```diff
@@ -0,0 +1,360 @@
+		/// attacking them passively.
+		/// </summary>
+		public void AttackBehavior()
+		{
+			if (HealTimer % 2 == 0 && life < MAX_LIFE)
+				life++;
+
+			// If we need to retarget, do so and reset the attack
+			if (!Projectile.TargetValid(800, true, true, target))
+			{
+				target = Projectile.TargetStrongestNPC(800, true, true);
+				Timer = 0;
+			}
+
+			// return to idle if no target
+			if (target is null)
+			{
+				State = 0;
+				Timer = 0;
+				return;
+			}
+
+			// Otherwise, we follow the attack cycle
+			if (Timer == 1)
+			{
+				startPoint = Projectile.Center;
+			}
+
+			// Position above the target
+			if (Timer > 1 && Timer < 30)
+			{
+				Projectile.velocity *= 0;
+				Vector2 targetPos = target.Center + Vector2.UnitY * (target.height * -0.5f - 64);
+				Projectile.Center = Vector2.SmoothStep(startPoint, targetPos, Timer / 30f);
+			}
+
+			// Fall onto them
+			if (Timer > 30 && Timer < 90)
+			{
+				Projectile.velocity.Y += 0.4f;
+				Projectile.Center = new Vector2(target.Center.X, Projectile.Center.Y);
+				Projectile.frame = 1;
+
+				// Bounce off the enemy once it's been stomped on
+				if (Projectile.Hitbox.Intersects(target.Hitbox))
+				{
+					Projectile.velocity.Y = -7;
+					Projectile.velocity.X = Main.rand.NextFloat(-5, 5);
+					Timer = 90;
+				}
+			}
+
+			// Continue to experience gravity, albeit reduced
+			if (Timer >= 90)
+			{
+				Projectile.velocity.Y += 0.2f;
+				Projectile.frame = 0;
+			}
+
+			// Fire a thorn at them
+			if (Timer == 110 && Main.netMode != NetmodeID.Server)
+			{
+				Projectile.NewProjectile(Projectile.GetSource_FromThis(), Projectile.Center, Projectile.Center.DirectionTo(target.Center + target.velocity * 10) * 11, ModContent.ProjectileType<SlimeThorn>(), 20, 0.5f, Projectile.owner);
+				SoundEngine.PlaySound(SoundID.DD2_DrakinShot, Projectile.Center);
+			}
+
+			// Do it all over again
+			if (Timer >= 120)
+				Timer = 0;
+		}
+
+		/// <summary>
+		/// The animation for when the minion is fusing with the player via the set bonus
+		/// </summary>
+		public void FuseAnimation()
+		{
+			if (Timer == 1)
+				startPoint = Projectile.Center;
+
+			if (Timer > 1 && Timer <= 60)
+				Projectile.Center = Vector2.SmoothStep(startPoint, Owner.Center + Vector2.UnitY * Owner.gfxOffY, Timer / 60f);
+
+			if (Timer == 60)
+			{
+				var helm = Owner.armor[0].ModItem as SlimePrinceHead;
+
+				if (helm != null)
+					helm.targetVel = new Vector2(0, -10);
+
+				State = 3;
+				Timer = 0;
+			}
+		}
+
+		/// <summary>
+		/// Behavior the minion should follow when it is fused to the player, via the set
+		/// bonus. This should stick hard to the player and constantly spawn spikes.
+		/// </summary>
+		public void FusedBehavior()
+		{
+			Projectile.velocity *= 0;
+			Projectile.Center = Owner.Center + Vector2.UnitY * Owner.gfxOffY;
+
+			life--;
+
+			Projectile.frame = Timer % 40 < 20 ? 1 : 0;
+
+			if (Timer % 30 == 0 && Main.netMode != NetmodeID.Server)
```

Comment:
same issue this needs to check against myplayer when spawning the projectile but this is good for the playsound part